controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/OsUpgradeScheduler.java

```diff
@@ -88,6 +88,14 @@ private boolean hasExpired(Version version) {
                            .collect(Collectors.toUnmodifiableSet());
     }
 
+    private boolean canTriggerAt(Instant instant) {
+        int hourOfDay = instant.atZone(ZoneOffset.UTC).getHour();
+        int dayOfWeek = instant.atZone(ZoneOffset.UTC).getDayOfWeek().getValue();
+        // Upgrade can only be scheduled between 07:00 and 12:59 UTC, Monday-Thursday
+        return hourOfDay >= 7 && hourOfDay <= 13 &&
```

Comment:
This is between 07:00 and 13:59? (Ref. comment above)

Reply:
Thanks, fixed.